portal-web/docroot/html/taglib/aui/input/page.jsp

```diff
@@ -25,39 +25,118 @@
 </c:if>
 
 <liferay-util:buffer
-	var="labelContent"
+	var="toggleSwitchLabelContent"
 >
 	<c:if test="<%= Validator.isNotNull(label) %>">
-		<c:if test='<%= type.equals("toggle-switch") %>'>
-			<span class="toggle-switch-label">
+		<span class="toggle-switch-label">
+			<liferay-ui:message key="<%= label %>" localizeKey="<%= localizeLabel %>" />
+		</span>
+
+		<c:if test="<%= required && showRequiredLabel %>">
+			<span class="toggle-switch-text">
+				<liferay-ui:message key="required" />
+
+				<aui:icon cssClass="reference-mark text-warning" image="asterisk" markupView="lexicon" />
+			</span>
 		</c:if>
+	</c:if>
+</liferay-util:buffer>
 
-		<liferay-ui:message key="<%= label %>" localizeKey="<%= localizeLabel %>" />
+<liferay-util:buffer
+	var="toggleSwitchStartContent"
+>
+	<label <%= labelTag %>>
+		<c:if test='<%= inlineLabel.equals("left") %>'>
+			<%= toggleSwitchLabelContent %>
+		</c:if>
 
-		<c:if test='<%= type.equals("toggle-switch") %>'>
+		<span aria-hidden="true" class="toggle-switch-check-bar">
+</liferay-util:buffer>
+
+<liferay-util:buffer
+	var="toggleSwitchEndContent"
+>
+
+			<%
+			String buttonIconOff = (String)dynamicAttributes.get("buttonIconOff");
+			String buttonIconOn = (String)dynamicAttributes.get("buttonIconOn");
+			String iconOff = (String)dynamicAttributes.get("iconOff");
+			String iconOn = (String)dynamicAttributes.get("iconOn");
+			String labelOff = (String)dynamicAttributes.get("labelOff");
+			String labelOn = (String)dynamicAttributes.get("labelOn");
+
+			if (localizeLabel) {
+				if (Validator.isNotNull(labelOff)) {
+					labelOff = LanguageUtil.get(resourceBundle, labelOff);
+				}
+
+				if (Validator.isNotNull(labelOn)) {
+					labelOn = LanguageUtil.get(resourceBundle, labelOn);
+				}
+			}
+			%>
+
+			<span class="toggle-switch-bar">
+				<span class="toggle-switch-handle" data-label-off="<%= Validator.isNotNull(labelOff) ? HtmlUtil.escapeAttribute(labelOff) : StringPool.BLANK %>" data-label-on="<%= Validator.isNotNull(labelOn) ? HtmlUtil.escapeAttribute(labelOn) : StringPool.BLANK %>">
+					<c:if test="<%= Validator.isNotNull(buttonIconOn) %>">
+						<span class="button-icon <%= Validator.isNotNull(buttonIconOff) ? "button-icon-on" : StringPool.BLANK %> toggle-switch-icon <%= buttonIconOn %>"></span>
+					</c:if>
+
+					<c:if test="<%= Validator.isNotNull(buttonIconOff) %>">
+						<span class="button-icon button-icon-off toggle-switch-icon <%= buttonIconOff %>"></span>
+					</c:if>
+
+					<c:if test="<%= Validator.isNotNull(iconOn) %>">
+						<span class="toggle-switch-icon toggle-switch-icon-on <%= iconOn %>"></span>
+					</c:if>
+
+					<c:if test="<%= Validator.isNotNull(iconOff) %>">
+						<span class="toggle-switch-icon toggle-switch-icon-off <%= iconOff %>"></span>
+					</c:if>
```

Comment:
There is a bit of markup duplication here. I assume we never want to render both off and on icons. Is there a variable we can leverage to know the state?

Even if not, we can do something like:
```
<span 
    class="
        toggle-switch-icon
        <%= Validator.isNotNull(iconOff) ? "toggle-switch-icon-off " + iconOff  : StringPool.BLANK %>
        <%= Validator.isNotNull(iconOn) ? "toggle-switch-icon-on " + iconOn  : StringPool.BLANK %>
    "
>
</span>
```

Reply:
No duplication here. The on icon is shown if the input is checked. The off icon is shown if the input is unchecked. This is pure CSS we aren't managing anything via JS.

Reply:
Duplicated markup is 
```
<span class="button-icon toggle-switch-icon"></span>
```
and 
```
<span class="toggle-switch-icon"></span>
```
The idea is to set the variable conditions to CSS classes, instead of the whole elements.

Reply:
I still don't see duplicates. We have variants [Toggle Switch with Icons](https://clayui.com/docs/components/toggle-switch/markup.html#css-with-icons). Not officially supported anymore, but left over from Clay 1.x.

Reply:
Sorry for not being clear enough. I don't mean that we are rendering duplicated markup. I mean that the markup is duplicated in code. For example, if we want to change the `button-icon` class, we need to do it in two places, on L82 and on L86. But, if we have something like
```
<span 
    class="
       button-icon toggle-switch-icon
        <%= Validator.isNotNull(buttonIconOff) ? "button-icon-off  " + buttonIconOff : Validator.isNotNull(buttonIconOn) ? "button-icon-on " + buttonIconOn  : StringPool.BLANK %>
    "
>
</span>
```
.. we only need to do it in one place. It's just a minor source formatting. This is not a blocker for PR, if you feel strongly about skipping DRY in this case, it's no problem to leave it as it is.

While typing the example above, I noticed that there is probably a bug on L82: `Validator.isNotNull(buttonIconOff)` check is very sus. It should probably just be `button-icon-on` instead.

Reply:
@markocikos we need all 4 span tags to be output if someone defines `buttonIconOn`, `buttonIconOff`, `iconOn`, and `iconOff`. Granted, I don't think this was ever working and used in portal.